### PR TITLE
auto select iptables mode

### PR DESCRIPTION
**Pull request by BenTheElder**
this follows equivalent logic to kube-proxy upstream. 

next step towards https://github.com/kubernetes-sigs/kind/issues/148, more PRs to follow ... (I'm cutting off smaller reasonable changes from my prototype)

### CODE REVIEW

**Comment by BenTheElder** [comment]
(pipefail wasn't enabled in another variant of this, updated)

File: images/base/files/usr/local/bin/entrypoint [hunk 1]
```diff
@@ -19,7 +19,7 @@ set -o nounset
 set -o pipefail
 
 fix_mount() {
-  echo 'INFO: ensuring we can execute /bin/mount even with userns-remap'
+  echo 'INFO: ensuring we can execute /bin/mount even with userns-remap'
```
Comment (aojea) [inline images/base/files/usr/local/bin/entrypoint]: Trailing whitespace?

**Comment by aojea** [comment]
Ijudy curious, is this a problem in kind?

File: images/base/files/usr/local/bin/entrypoint (hunk 1 above)
Comment (BenTheElder) [reply images/base/files/usr/local/bin/entrypoint]: 🤷‍♂️ this won't be the last chance to this file and we bumped the image

**Comment by BenTheElder** [comment]
Not yet, no.

**Pull request merged**